flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/config/ExecutionConfigOptions.java

```diff
@@ -83,97 +83,97 @@
-			key("sql.resource.default.parallelism")
+	public static final ConfigOption<Integer> TABLE_EXEC_RESOURCE_DEFAULT_PARALLELISM =
+			key("table.exec.resource.default-parallelism")
 					.defaultValue(-1)
 					.withDescription("Default parallelism of job operators. If it is <= 0, use parallelism of StreamExecutionEnvironment(" +
 							"its default value is the num of cpu cores in the client host).");
 
-	public static final ConfigOption<Integer> SQL_RESOURCE_SOURCE_PARALLELISM =
-			key("sql.resource.source.parallelism")
+	public static final ConfigOption<Integer> TABLE_EXEC_RESOURCE_SOURCE_PARALLELISM =
+			key("table.exec.resource.source.parallelism")
 					.defaultValue(-1)
-					.withDescription("Sets source parallelism, if it is <= 0, use " + SQL_RESOURCE_DEFAULT_PARALLELISM.key() + " to set source parallelism.");
+					.withDescription("Sets source parallelism, if it is <= 0, use " + TABLE_EXEC_RESOURCE_DEFAULT_PARALLELISM.key() + " to set source parallelism.");
 
-	public static final ConfigOption<Integer> SQL_RESOURCE_SINK_PARALLELISM =
-			key("sql.resource.sink.parallelism")
+	public static final ConfigOption<Integer> TABLE_EXEC_RESOURCE_SINK_PARALLELISM =
+			key("table.exec.resource.sink.parallelism")
 					.defaultValue(-1)
-					.withDescription("Sets sink parallelism, if it is <= 0, use " + SQL_RESOURCE_DEFAULT_PARALLELISM.key() + " to set sink parallelism.");
+					.withDescription("Sets sink parallelism, if it is <= 0, use " + TABLE_EXEC_RESOURCE_DEFAULT_PARALLELISM.key() + " to set sink parallelism.");
 
-	public static final ConfigOption<Integer> SQL_RESOURCE_EXTERNAL_BUFFER_MEM =
-			key("sql.resource.external-buffer.memory.mb")
-					.defaultValue(10)
+	public static final ConfigOption<String> TABLE_EXEC_RESOURCE_EXTERNAL_BUFFER_MEM =
+			key("table.exec.resource.external-buffer.memory")
+					.defaultValue("10 mb")
 					.withDescription("Sets the externalBuffer memory size that is used in sortMergeJoin and overWindow.");
 
-	public static final ConfigOption<Integer> SQL_RESOURCE_HASH_AGG_TABLE_MEM =
-			key("sql.resource.hash-agg.table.memory.mb")
-					.defaultValue(128)
-					.withDescription("Sets the table memory size of hashAgg operator.");
+	public static final ConfigOption<String> TABLE_EXEC_RESOURCE_HASH_AGG_TABLE_MEM =
+			key("table.exec.resource.hash-agg.memory")
+					.defaultValue("128 mb")
+					.withDescription("Sets the managed memory size of hash aggregate operator.");
 
-	public static final ConfigOption<Integer> SQL_RESOURCE_HASH_JOIN_TABLE_MEM =
-			key("sql.resource.hash-join.table.memory.mb")
-					.defaultValue(128)
-					.withDescription("Sets the HashTable reserved memory for hashJoin operator. It defines the lower limit.");
+	public static final ConfigOption<String> TABLE_EXEC_RESOURCE_HASH_JOIN_TABLE_MEM =
+			key("table.exec.resource.hash-join.memory")
+					.defaultValue("128 mb")
+					.withDescription("Sets the managed memory for hash join operator. It defines the lower limit.");
```

Comment:
nit: could you also update the typos in other descriptions such as `externalBuffer` -> `external buffer`, `sortMergeJoin` -> `sort-merge join` etc.